pkg/kubelet/volumes.go

```diff
@@ -96,6 +96,34 @@ func (kl *Kubelet) newVolumeBuilderFromPlugins(spec *volume.Spec, pod *api.Pod,
 	return builder, nil
 }
 
+func (kl *Kubelet) setVolumeOwnership(volume api.Volume, podRef *api.Pod) error {
+	// Determine volume usage
+	containers := []api.Container{}
+
+	for _, container := range pod.Spec.Containers {
+		for _, volumeMount := range container.VolumeMounts {
+			if volumeMount.Name == volume.Name {
+				containers = append(containers, container)
+				break
+			}
+		}
+	}
+
+	if len(containers) == 0 {
+		return nil
+	}
+
+	// Do images need to be pulled here?
```

Comment:
For the record, I think this will need to be a function of the container runtime, because the image pull happens behind a `Runtime` interface method call.